### DISCUSSION TITLE
How setup FLEX?

### DISCUSSION THREAD

**Discussion by toysonlocation** [airsdk/Adobe-Runtime-Support#1861] [Q&A] [answered]
Sorry for my dumb question: I've always developed pure AS3 apps, and now I have to help fix a FLEX application. How do I get/setup the FLEX SDK? I'm using IntelliJ. I've found an old copy of "Flex 4.6" on one of my drives, copied the contents of the latest Harman AIR SDK (821) into it, and am getting the following error:

[app-dev (module app)]: java.net.SocketTimeoutException: Accept timed out
[app-dev (module app)]: Exception in thread "main" 
[app-dev (module app)]: java.lang.NoClassDefFoundError: flex2/tools/VersionInfo

My Flex/bin directory DOES contain mxmlc

--------
UPDATED: Here is the command that ends in the error:

[app-dev (module app)]: Starting Flex compiler:
"/Applications/IntelliJ IDEA.app/Contents/jbr/Contents/Home/bin/java" -Dapplication.home=/Volumes/Tub/Resources/Adobe/AIR/FLEX_SDK_4.6_AIR_33.1.1.821_overlay -Dfile.encoding=UTF-8 -Djava.awt.headless=true -Duser.language=en -Duser.region=en -Xmx512m -classpath "/Users/work/Library/Application Support/JetBrains/IntelliJIdea2021.1/plugins/flex/lib/idea-flex-compiler-fix.jar:/Users/work/Library/Application Support/JetBrains/IntelliJIdea2021.1/plugins/flex/lib/flex-compiler.jar:/Volumes/Tub/Resources/Adobe/AIR/FLEX_SDK_4.6_AIR_33.1.1.821_overlay/lib/flex-compiler-oem.jar" com.intellij.flex.compiler.FlexCompiler 49300

**Comment by toysonlocation** [comment, marked as answer]
OK I've fixed the above. After numerous failed attempts to merge the Apache FLEX 4.16.1 with Harman AIR 33.1.1.821 (funny the lack of "Adobe" in all that) I got it working. I used the "ditto" command in bash on macOS, as in...

% ditto PATH_TO_AIR_SDK PATH_TO_FLEX_SDK

At least, mostly fixed. New post with different issue coming up.

**Reply by ajwfrost** [reply]
Interesting, I didn't recognise the error displayed there .. but yes in terms of overlaying the SDKs, a normal copy in macOS would replace the full contents of a folder where the folder names match, so you need to use 'ditto'. We're looking at providing a better way to do this sort of task......
